src/index.js

```diff
@@ -0,0 +1,150 @@
+    "@typescript-eslint/tslint",
+    "react",
+    "cypress",
+    "jest",
+    "testing-library",
+    "prettier",
+    "import",
+    "json",
+  ],
+  settings: {
+    react: {
+      version: "detect",
+    },
+  },
+  overrides: [
+    {
+      files: ["*.js", "*.jsx"],
+      rules: {
+        "@typescript-eslint/no-var-requires": "off",
+        "@typescript-eslint/explicit-function-return-type": "off",
+      },
+    },
+    {
+      files: ["**/cypress/**/*"],
+      rules: {
+        "jest/expect-expect": "off",
+        "jest/valid-expect-in-promise": "off",
+        "jest/valid-expect": "off",
+        "jest/no-standalone-expect": "off",
+        "testing-library/await-async-query": "off",
+      },
+    },
+    {
+      files: [
+        "**/__tests__/next/**/*.test.tsx",
+        "**/__tests__/pages/**/*.test.tsx",
+        "**/__tests__/pages/**/*.test.jsx",
+        "**/components/**/*.test.tsx",
+        "**/components/**/*.test.jsx",
+      ],
+      rules: {
+        // off because of testing library
+        "jest/valid-expect": "off",
+        "jest/expect-expect": "off",
+      },
+    },
+    {
+      files: [
+        "babel.config.js",
+        "jest.config.js",
+        "release.config.js",
+        ".eslintrc.js",
+        "*.json",
+      ],
+      parser: "esprima",
+    },
+  ],
+  rules: {
+    "@typescript-eslint/explicit-function-return-type": "off",
+    "@typescript-eslint/no-unused-vars": [
+      "error",
+      {
+        varsIgnorePattern: "^_",
+        argsIgnorePattern: "^_",
+        ignoreRestSiblings: true,
+      },
+    ],
+    "react/prop-types": "off",
+    "react/display-name": "off",
+    "jest/no-mocks-import": "off",
+    "prettier/prettier": ["error", { semi: false }],
+    "prefer-spread": "off",
+    "no-console": ["error", { allow: ["info", "warn", "error"] }],
+    "no-irregular-whitespace": [
+      "error",
+      {
+        skipStrings: true,
+        skipTemplates: true,
+      },
+    ],
+    "no-restricted-imports": [
+      "error",
+      {
+        paths: [
+          {
+            name: "next/link",
+            message: 'Please import from "~/components/appLink.tsx" instead.',
+          },
+          {
+            name: "next/router",
+            importNames: ["default"],
+            message: 'Please import from "~/server/routing/routes.ts" instead.',
+          },
+        ],
+      },
+    ],
+    // TODO: we had that enabled but also disabled on in multiple files
+    "sort-keys": "off",
+    "import/no-internal-modules": [
+      "error",
+      {
+        allow: [
+          "**/*.css",
+          "**/configuration/**/*",
+          "**/configuration/*",
+          "**/__fixtures__/**/*",
+          "**/__fixtures__/*",
+          "@testing-library/**/*",
+          "next/*",
+          "next-i18next/*",
+          "react-dom/*",
+        ],
+      },
+    ],
+    "json/*": ["error", "allowComments"],
+    "react-hooks/exhaustive-deps": "off",
```

Comment:
Found this https://reactjs.org/docs/hooks-rules.html
But did not fully understand what it means, can you explain it a bit

Reply:
It's a relatively new rule that wants you to declare all variables you reference inside `useEffect` in the dependencies array. We have a bunch of useEffects that depend on `id` only even though they reference the full object and it would force us to add the object to the dependency array making the effect always trigger.

Reply:
So it's like where we use the deepComare function to trigger the hook for changes deeper in the object?

Reply:
Kind of, but also not exactly. Take this snipper for instance:
```
useEffect(() => {
    const updateProfile = async () => {
      const newProfile = await fetchDealerProfile(activeDealerId)
      if (deepEqual(newProfile, profile)) {
        setDealerProfile(newProfile)
      }
    }

    if (activeDealerId && hasRole(currentUser, Role.PrivateSeller)) {
      updateProfile()
    }
  }, [activeDealerId])
```

what the rule would suggest is to add:
`profile` and `currentUser` to the dependency array.
1. This would require us to switch to deep compare since we're using objects
2. In this particular case was want to only depend on the ID to re-etch data

Reply:
Thanks for the explanation 😊, I see how the rule can help with not missing cases/forces you to think about what triggers change the result of you state. Im fine with initialy exlcuding it, let's consider looking in to it with the team.